### PR TITLE
added validation information

### LINKED ISSUE #171: Validate task data against dependency data

### PR DESCRIPTION
**Pull request by SamMarkowitz**
fixes #171 

This will probably have conflicts by the time it is being merged if after #174. I will fix then.

### CODE REVIEW

**Comment by levice14** [comment]
@SamMarkowitz please resolve conflicts with master

File: cloudslang_dsl_reference.rst [hunk 1]
```diff
@@ -850,16 +850,21 @@ For more information, see the :ref:`Operation Paths <example_operation_paths>`
...
+value from a variable with the same name in the flow context.
+`Expression <#expressions>`__ values will supersede values bound to flow
+`inputs <#inputs>`__ with the same name. To force the `operation <#operation>`__
+or `subflow <#flow>`__ being called to use it's own default value, as opposed to
+a value passed in via expression or the flow context, omit the variable from the
+calling `step's <#step>`__ argument list.
 
 For a list of which contexts are available in the arguments section of a
 `step <#step>`__, see `Contexts <#contexts>`__.
+>>>>>>> master
```
Comment (levice14) [inline cloudslang_dsl_reference.rst]: naughty Git - please take a look @SamMarkowitz

Comment (SamMarkowitz) [inline cloudslang_dsl_reference.rst]: Harumph. My fault, not Git's.

**Comment by levice14** [comment]
:ok:

**Pull request merged**